### PR TITLE
GH-1044 Improve Settings API

### PR DESCRIPTION
**Pull request by Minecraftschurli**
Related issues:
- [x] #1044 
  - [x] Forms scheme
  - [x] Forms UI
  - [x] styling
- [x] #1150 
- [x] #1151 
- [x] https://github.com/dzikoysk/reposilite/issues/1176
- [x] https://github.com/dzikoysk/reposilite/issues/1170
- [x] #1191
- [x] https://github.com/dzikoysk/reposilite/issues/1026
- [x] Manual management of shared configuration

### CODE REVIEW

File: reposilite-backend/src/main/kotlin/com/reposilite/settings/api/GeneralSettings.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.reposilite.settings.api
+
+import java.io.Serializable
+
+data class GeneralSettings(
```
Comment (dzikoysk) [inline reposilite-backend/src/main/kotlin/com/reposilite/settings/api/GeneralSettings.kt:5]: Maybe sth like `AppearanceSettings`

**Comment by dzikoysk** [comment]
I've made 2 new issues that address problems I see here. I'm quite sure it'll simplify the whole related codebase and we can avoid a lot of unnecessary conversions as I don't quite like it atm 🤔 I'm currently focused on #1141, but then I'll be able to help here if needed.

File: reposilite-backend/src/main/kotlin/com/reposilite/settings/application/SettingsPlugin.kt [hunk 2]
```diff
@@ -42,7 +51,33 @@ internal class SettingsPlugin : ReposilitePlugin() {
 
         val database = DatabaseSourceFactory.createConnection(workingDirectory, localConfiguration.database.get())
         val settingsRepository = SqlSettingsRepository(database)
-        val settingsFacade = SettingsFacade(this, workingDirectory, localConfiguration, lazy { database }, settingsRepository)
+
+        // TODO: Move to storage domain or remove this kind of impl if possible
```
Comment (dzikoysk) [inline reposilite-backend/src/main/kotlin/com/reposilite/settings/application/SettingsPlugin.kt:55]: .

**Comment by EthanDevelops** [comment]
Any ETA on this? No rush though

**Comment by EthanDevelops** [comment]
Any update on this?

**Comment by dzikoysk** [comment]
I'd say that the core functionality is pretty much done! We need to provide a few side features such as factory reset & a possibility to mount settings as file.

**Review by dzikoysk — APPROVED** [review]

**Comment by dzikoysk** [comment]
I'll probably merge this tomorrow, there are only some small minor issues I'd like to adjust before merge.

**Comment by dzikoysk** [comment]
Todo:
* [ ] Generate default shared configuration file through command
* [ ] Save shared configuration after update

**Comment by dzikoysk** [comment]
JsonForms 3.0.0-beta.2 removed `resolveSubSchemas`, so we need to replace it somehow. Changes:

https://github.com/eclipsesource/jsonforms/compare/v3.0.0-beta.1...v3.0.0-beta.2

**Comment by dzikoysk** [comment]
Thanks for your contribution @Minecraftschurli and all your time you've spend we me to get it done! ❤️

**Pull request merged**